### PR TITLE
More github actions updates

### PR DESCRIPTION
**Pull request by busterb**
Following up on work from @kinichiro, this splits out a few more tests into logical actions so we can do more testing in parallel. It also renames the testing script and adds updated badges to the README.md.

### CODE REVIEW

File: .github/workflows/cross_test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+        arch: [mingw32, mingw64, arm32, arm64]
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    env:
+      CC: gcc
+      ARCH: ${{ matrix.arch }}
+    name: ${{ matrix.arch }}
```
Comment (kinichiro) [inline .github/workflows/cross_test.yml:16]: I would like to add ${{ matrix.os }} to the name to see if it is 18 or 20.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
 ## Official portable version of [LibreSSL](https://www.libressl.org) ##
 
-[![Build Status](https://travis-ci.org/libressl-portable/portable.svg?branch=master)](https://travis-ci.org/libressl-portable/portable) [![Fuzzing Status](https://oss-fuzz-build-logs.storage.googleapis.com/badges/libressl.svg)](https://bugs.chromium.org/p/oss-fuzz/issues/list?sort=-opened&can=1&q=proj:libressl)
+[![Linux Build Status](https://github.com/libressl-portable/portable/actions/workflows/linux_test.yml/badge.svg)](https://github.com/libressl-portable/portable/actions/workflows/linux_test.yml)
+[![macOS Build Status](https://github.com/libressl-portable/portable/actions/workflows/macos_test.yml/badge.svg)](https://github.com/libressl-portable/portable/actions/workflows/macos_test.yml)
+[![Android_Build Status](https://github.com/libressl-portable/portable/actions/workflows/android_test.yml/badge.svg)](https://github.com/libressl-portable/portable/actions/workflows/android_test.yml)
+[![Cross_Build Status](https://github.com/libressl-portable/portable/actions/workflows/cross_test.yml/badge.svg)](https://github.com/libressl-portable/portable/actions/workflows/cros_test.yml)
```
Comment (kinichiro) [inline README.md:7]: typo cros

**Comment by kinichiro** [comment]
Great and I think it is good to go in this.
And we found another issues to solve on ubuntu 20 with cross build.

File: .github/workflows/cross_test.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+on: [push, pull_request]
+
+jobs:
+  build-other:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+        arch: [mingw32, mingw64, arm32, arm64]
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
```
Comment (kinichiro) [inline .github/workflows/cross_test.yml:12]: This continue-on-error should be false to align with other workflows.

**Comment by kinichiro** [comment]
I think #674 would fix mingw build failure on ubuntu 20.04.
I found arm toolchain on ubuntu appears to be changed a little bit between 18 and 20,
and ld file version had been updated from 2.27 to 2.31.
Also, it require to provide the path for libpthread, then I added LD_LIBRARY_PATH to show it.
I believe diff below would fix that.
```diff
diff --git a/scripts/test b/scripts/test
index a18f58c..80fa988 100755
--- a/scripts/test
+++ b/scripts/test
@@ -81,14 +81,14 @@ elif [ "x$ARCH" = "xarm32" -o "x$ARCH" = "xarm64" ]; then

        if [ "x$ARCH" = "xarm32" ]; then
                sudo apt-get install -y g++-arm-linux-gnueabihf
-               sudo ln -s /usr/arm-linux-gnueabihf/lib /lib/arm-linux-gnueabihf
-               sudo ln -s /lib/arm-linux-gnueabihf/ld-2.27.so /lib/ld-linux-armhf.so.3
+               sudo ln -s /usr/arm-linux-gnueabihf/lib/ld-*.*.so /lib/ld-linux-armhf.so.3
+               export LD_LIBRARY_PATH=/usr/arm-linux-gnueabihf/lib
                export CC=arm-linux-gnueabihf-gcc
                ./configure --host=arm-linux
        else
                sudo apt-get install -y g++-aarch64-linux-gnu
-               sudo ln -s /usr/aarch64-linux-gnu/lib/ /lib/aarch64-linux-gnu
-               sudo ln -s /lib/aarch64-linux-gnu/ld-2.27.so /lib/ld-linux-aarch64.so.1
+               sudo ln -s /usr/aarch64-linux-gnu/lib/ld-*.*.so /lib/ld-linux-aarch64.so.1
+               export LD_LIBRARY_PATH=/usr/aarch64-linux-gnu/lib
                export CC=aarch64-linux-gnu-gcc
                ./configure --host=aarch64-linux
        fi
```

**Comment by busterb** [comment]
Thanks @kinichiro, I landed your fixes and rebased on those.

**Pull request merged**